### PR TITLE
Add constant spacing

### PR DESCRIPTION
**Pull request by SaraVieira**
This PR adds constant spacing throughout the project with an 8px grid.

I did this by adding the spacing variables to the theme like so:

```
const spacingFactor = 8;
export const spacing = {
	space4: `${spacingFactor / 2}px`,  // 4
	space8: `${spacingFactor}px`,      // 8
	space16: `${spacingFactor * 2}px`,  // 16
	space24: `${spacingFactor * 3}px`,  // 24
	space32: `${spacingFactor * 4}px`,  // 32
	space40: `${spacingFactor * 5}px`,  // 40
	space48: `${spacingFactor * 6}px`,  // 48
};
```

These were the only spaces I used in the entire project to main consistency.

### CODE REVIEW

File: src/rsg-components/Heading/HeadingRenderer.js [hunk 1]
```diff
@@ -3,22 +3,23 @@ import PropTypes from 'prop-types';
...
-export const styles = ({ base, link, linkHover }) => ({
+export const styles = ({ base, link, linkHover, spacing }) => ({
 	heading: {
 		color: base,
 		position: 'relative',
 		overflow: 'visible',
-		marginLeft: -30,
-		paddingLeft: 30,
+		marginLeft: -32,
+		paddingLeft: spacing.space32,
```
Comment (sapegin) [inline src/rsg-components/Heading/HeadingRenderer.js]: I think they both should you the same variable.

File: src/rsg-components/Methods/Methods.css [hunk 2]
```diff
@@ -9,19 +9,19 @@
 .tableBody {
```
Comment (sapegin) [inline src/rsg-components/Methods/Methods.css]: We don’t use this file, could you please remove it? Merge artifact ;-/

File: src/rsg-components/Methods/MethodsRenderer.js [hunk 3]
```diff
@@ -20,24 +20,24 @@ const styles = ({ base, font, border, light, name, type }) => ({
...
 		verticalAlign: 'top',
 		fontFamily: font,
 		fontSize: 13,
 	},
 	cellHeading: {
 		color: base,
-		paddingRight: 15,
-		paddingBottom: 6,
+		paddingRight: spacing.space16,
+		paddingTop: spacing.space8,
```
Comment (sapegin) [inline src/rsg-components/Methods/MethodsRenderer.js]: Looks like a typo, it should be bottom padding.

File: src/rsg-components/PlaygroundError/PlaygroundErrorRenderer.js [hunk 4]
```diff
@@ -2,13 +2,13 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import Styled from 'rsg-components/Styled';
 
-const styles = ({ monospace, white, errorBackground }) => ({
+const styles = ({ monospace, white, errorBackground, spacing }) => ({
 	root: {
-		margin: [[-15, -15, -18]],
+		margin: [[-16, -16, -16]],
```
Comment (sapegin) [inline src/rsg-components/PlaygroundError/PlaygroundErrorRenderer.js]: space16? But actually this numbers are not random — the red background should fill the playground container.

File: src/styles/theme.js [hunk 5]
```diff
@@ -10,7 +12,17 @@ export const type = '#b77daa';
 export const baseBackground = '#fff';
 export const errorBackground = '#c00';
 export const codeBackground = '#f5f5f5';
-export const font = ['-apple-system', 'BlinkMacSystemFont', '"Segoe UI"', '"Roboto"', '"Oxygen"', '"Ubuntu"',
+export const font = ['"Roboto"', '-apple-system', 'BlinkMacSystemFont', '"Segoe UI"', '"Roboto"', '"Oxygen"', '"Ubuntu"',
```
Comment (sapegin) [inline src/styles/theme.js]: Why Roboto?

File: src/styles/theme.js [hunk 6]
```diff
@@ -10,7 +12,17 @@ export const type = '#b77daa';
...
 	'"Cantarell"', '"Fira Sans"', '"Droid Sans"', '"Helvetica Neue"', 'sans-serif'];
 export const monospace = ['Consolas', '"Liberation Mono"', 'Menlo', 'monospace'];
 export const small = '@media (max-width: 600px)';
+export const sidebarWidth = 200;
+export const spacing = {
```
Comment (sapegin) [inline src/styles/theme.js]: I’d use `space.0`, `space.1`, `space.2`. Actual numbers are implementation detail, so it’s a bit like `red = blue` thing ;-)

File: src/rsg-components/ComponentsList/ComponentsListRenderer.js [hunk 7]
```diff
@@ -4,30 +4,30 @@ import cx from 'classnames';
...
+		paddingLeft: spacing.space16,
 	},
 	item: {
 		color: base,
 		display: 'block',
-		margin: [[7, 0, 7, 0]],
+		margin: [[spacing.space8, 0, spacing.space8, 0]],
 		fontFamily: font,
-		fontSize: 15,
+		fontSize: spacing.space16,
```
Comment (sapegin) [inline src/rsg-components/ComponentsList/ComponentsListRenderer.js]: We shouldn’t use spacers for font sizes. But would be nice to have variables for that too.

File: src/rsg-components/Methods/Methods.css (hunk 2 above)
Comment (SaraVieira) [inline src/rsg-components/Methods/Methods.css]: Sure!

File: src/rsg-components/Methods/MethodsRenderer.js (hunk 3 above)
Comment (SaraVieira) [inline src/rsg-components/Methods/MethodsRenderer.js]: Sorry , typo

File: src/styles/theme.js (hunk 5 above)
Comment (SaraVieira) [inline src/styles/theme.js]: Because it's pretty. That's all :p

File: src/rsg-components/PlaygroundError/PlaygroundErrorRenderer.js (hunk 4 above)
Comment (SaraVieira) [inline src/rsg-components/PlaygroundError/PlaygroundErrorRenderer.js]: I will change them back and add that as a comment

File: src/styles/theme.js (hunk 6 above)
Comment (SaraVieira) [inline src/styles/theme.js]: I did it like this to be easier to make sense of it, so when someone sees space16 they will see its 16 px

File: src/rsg-components/ComponentsList/ComponentsListRenderer.js (hunk 7 above)
Comment (SaraVieira) [inline src/rsg-components/ComponentsList/ComponentsListRenderer.js]: Typo again, but I can add some :)

File: src/styles/theme.js (hunk 5 above)
Comment (sapegin) [inline src/styles/theme.js]: We’re using system fonts.

File: src/styles/theme.js (hunk 6 above)
Comment (sapegin) [inline src/styles/theme.js]: In that case you need to remember that and calculate the right number every time, with 1, 2, 3 it easy: single space, double space…

File: src/styles/theme.js (hunk 5 above)
Comment (SaraVieira) [inline src/styles/theme.js]: But why ?

File: src/styles/theme.js (hunk 6 above)
Comment (SaraVieira) [inline src/styles/theme.js]: But the first one is half , how should I call it ?

**Comment by SaraVieira** [comment]
@sapegin Most of the requested changes were made and I added font variables too :)

File: src/styles/theme.js (hunk 6 above)
Comment (SaraVieira) [inline src/styles/theme.js]: Spacing renamed to: 

```
export const spacing = {
	half: `${spacingFactor / 2}px`,  // 4
	1: `${spacingFactor}px`,      // 8
	2: `${spacingFactor * 2}px`,  // 16
	3: `${spacingFactor * 3}px`,  // 24
	4: `${spacingFactor * 4}px`,  // 32
	5: `${spacingFactor * 5}px`,  // 40
	6: `${spacingFactor * 6}px`,  // 48
};
```

Comment (SaraVieira) [inline src/styles/theme.js]: I also exported the spacingFactor if someone wants to change it in their project

Comment (sapegin) [inline src/styles/theme.js]: I’d call it 0: https://medium.freecodecamp.com/css-in-javascript-the-future-of-component-based-styling-70b161a79a32

File: src/styles/theme.js (hunk 5 above)
Comment (sapegin) [inline src/styles/theme.js]: They are available on all platforms, they look good (especially SF ;-) and they are used in the OS UI. 

How many people have Roboto installed? :-)

File: src/styles/theme.js (hunk 6 above)
Comment (SaraVieira) [inline src/styles/theme.js]: Changed :)

File: src/styles/theme.js (hunk 5 above)
Comment (SaraVieira) [inline src/styles/theme.js]: Not many , but for those who don't we can just go to the apple-system ones

Comment (sapegin) [inline src/styles/theme.js]: Sorry, I don’t see any reason to use Roboto just for a few people who have it, while system fonts are still better.

File: src/styles/theme.js (hunk 6 above)
Comment (sapegin) [inline src/styles/theme.js]: 👍

**Comment by SaraVieira** [comment]
@sapegin All changed made ? may I merge ?

**Comment by sapegin** [comment]
Merged, thanks!

**Comment by sapegin** [comment]
I’ve made some tweaks in 19ed879 ;-)

**Pull request merged**